docs/DeveloperGuide.md

```diff
@@ -767,42 +767,186 @@ testers are expected to do more *exploratory* testing.
 
    1. Resize the window to an optimum size. Move the window to a different location. Close the window.
 
-   1. Re-launch the app by double-clicking the jar file.<br>
+   2. Re-launch the app by double-clicking the jar file.<br>
        Expected: The most recent window size and location is retained.
 
-1. _{ more test cases …​ }_
+### Contact Management
+   
+#### Adding a contact
 
-### Deleting a contact
+1. Adding a new contact.
+   1. Prerequisite: There is no contact named `John Doe`.
+   2. Test case: `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01`.
+   3. Expected: New contact with the same details added in the contact list.
+2. Adding a contact that has existed before.
+   1. Prerequisite: There is a contact named `Betsy Crowe` in the contact list.
+   2. Text case: `add n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567`.
+   3. Expected: Error message `This person already exists in the SoConnect.` is displayed.
+   
+#### Editing a contact
 
-1. Deleting a contact while all contacts are being shown
+1. Editing an existing contact.
+   1. Prerequisite: There is a contact in the 2nd index of contact list.
+   2. Test case: `edit 2 n/Betsy Crower`.
+   3. Expected: Edits the name of the 2nd contact to be Betsy Crower.
+   
+#### Listing all contacts
+
+1. Prerequisite: There are some existing contacts.
+2. Test case: `list`.
+3. Expected: Display all contacts.
+
+#### Searching for a contact
+
+1. Searching an existing contact.
+   1. Prerequisite: There is a contact named `Susan` in the contact list.
+   2. Test case: `search n/susan`.
+   3. Expected: Display all contacts that contain `susan` as name, including `Susan`.
+
+#### Sorting contacts
+
+1. Prerequisite: There are some existing contacts.
+2. Test case: `sort n/`.
+3. Expected: Display all contacts with the names sorted from A - Z.
+
+#### Deleting a contact
+
+1. Deleting a contact while all contacts are being shown.
 
    1. Prerequisites: List all contacts using the `list` command. Multiple contacts in the list.
 
-   1. Test case: `delete 1`<br>
+   1. Test case: `delete 1`.<br>
       Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
 
-   1. Test case: `delete 0`<br>
+   1. Test case: `delete 0`.<br>
       Expected: No contact is deleted. Error details shown in the status message. Status bar remains the same.
 
    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
 
-1. _{ more test cases …​ }_
+#### Clearing all contacts
+
+1. Prerequisite: There are some existing contacts.
+2. Test case: `clear`.
+3. Expected: All contacts are erased.
+
+### Todo Management
+
+#### Adding a todo
+
+1. Adding a new todo.
+   1. Test case: `todo add d/Watched recorded videos for CS2100 date/24-10-2022 pr/low t/CS2100`.
```

Comment:
Prereq: CS2100 tag must be created beforehand and no existing todo has the exact same details